Motion/app/src/main/java/com/example/android/motion/demo/containertransform/CheeseArticleFragment.kt

```diff
@@ -111,5 +118,45 @@ class CheeseArticleFragment : Fragment() {
         toolbar.setNavigationOnClickListener { v ->
             v.findNavController().popBackStack()
         }
+
+        requireActivity().onBackPressedDispatcher.addCallback(
+            viewLifecycleOwner,
+            object : OnBackPressedCallback(true) {
+                override fun handleOnBackPressed() {
+                    // This invokes the sharedElementReturnTransition, which is
+                    // MaterialContainerTransform.
+                    // TODO: It pops the scaleX/Y back to 1f. It would look nicer to retain the
```

Comment:
It's the transition after the back gesture is committed (e.g. MaterialContainerTransform for the sharedElementReturnTransition) that is popping the scale back to 1f?

Reply:
Shared elements are copied into the target fragment before the transition animations start. It seems the scale and other transformations are dropped in the process. I tried using a set of `ChangeTransform` (it handles scaleX/Y) and `MaterialContainerTransform`, but it doesn't work for some reason.

I updated the formula below, and this is hardly unnoticeable. I think I don't care about this for now.

Reply:
with the updated formulas the reset is not very noticeable at all

thanks Yuichi!